rust_src/src/windows.rs

```diff
@@ -2017,4 +2059,34 @@ pub fn window_pixel_height_before_size_change(window: LispWindowValidOrSelected)
     window.pixel_height_before_size_change
 }
 
+/// Get width and type of scroll bars of window WINDOW.
+/// WINDOW must be a live window and defaults to the selected one.
+///
+/// Value is a list of the form (WIDTH COLUMNS VERTICAL-TYPE HEIGHT LINES
+/// HORIZONTAL-TYPE).  If WIDTH or HEIGHT is nil or VERTICAL-TYPE or
+/// HORIZONTAL-TYPE is t, the window is using the frame's corresponding
+/// value.
+#[lisp_fn(min = "0")]
+pub fn window_scroll_bars(window: LispWindowLiveOrSelected) -> LispObject {
+    let window: LispWindowRef = window.into();
+    let width = if window.scroll_bar_width >= 0 {
+        Some(window.scroll_bar_width)
+    } else {
+        None
+    };
+    let height = if window.scroll_bar_height >= 0 {
+        Some(window.scroll_bar_height)
+    } else {
+        None
+    };
+    list!(
```

Comment:
Why did the original use `Fcons` instead of `list6()`?

Reply:
I believe that they only have up to `list5`: https://github.com/remacs/remacs/blob/fb2e5d73bbd221568e9fbeaa2e6d169a422455de/src/lisp.h#L3706-L3711

With `list!`, I think the rust version really shines

Reply:
Agreed. `list!` is so much nicer. I suspected that was the answer about list5, but I did not have the chance to check.